### DISCUSSION TITLE
Can I automatically convert all C# files to use file-scoped namespaces introduced in C#10?

### DISCUSSION THREAD

**Discussion by MartyIX** [dotnet/roslyn#56712] [Q&A] [answered]
Hi,

I have a project where most of the code can be converted to use C# 10 file-scope namespaces. Is there a code-fixer to do this? It seems the feature would be very useful for many people if there isn't.

Thank you

**Comment by xqrzd** [comment, marked as answer]
In VS 2022 open options, and go to Text Editor -> C# => Code Style => General
Change Namespace declarations to file scoped,

![image](https://user-images.githubusercontent.com/4950876/134733236-5d989352-c2af-4836-906f-d0c161e90c34.png)

Then open one of your .cs files, open quick actions on the namespace, and select convert to file-scoped namespace. It will have options at the bottom to convert the entire project/solution.

![image](https://user-images.githubusercontent.com/4950876/134733878-4533861f-dfcf-4583-8329-3f2d5a6be2ab.png)

**Reply by MartyIX** [reply]
Thanks!
